src/invoice.ts

```diff
@@ -0,0 +1,36 @@
+import fetch from 'cross-fetch';
+import { getHashFromInvoice } from "./utils/invoice";
+import Hex from "crypto-js/enc-hex";
+import sha256 from "crypto-js/sha256";
+
+export default class Invoice {
+  paymentRequest: string;
+  paymentHash: string;
+  verify: string;
+
+  constructor(lnUrlPayResponse: {pr: string, verify: string}) {
+    this.paymentRequest = lnUrlPayResponse.pr;
+    this.verify = lnUrlPayResponse.verify;
+    this.paymentHash = getHashFromInvoice(this.paymentRequest) as string;
```

Comment:
let's also allow to pass in the preimage here. 

then we can do: 

```
const invoice = new Invoice(params);

if (invoice.isPaid()) {
  ...
}
```

the `isPaid()` function then either checks: 
1. preimage with the payment hash
2. the verify URL
3. thows an error if both does not work

Reply:
Done!